lokimq/bt_serialize.h

```diff
@@ -909,5 +922,192 @@ class bt_dict_consumer : private bt_list_consumer {
     bt_dict_consumer consume_dict_consumer() { return consume_dict_data(); }
 };
 
+/// Currently this is not idiot-proof.  That is to say, the compiler won't catch everything and
+/// you technically can create invalid BEncoded strings.  Basically you can add list elements
+/// to a dict and dict elements to a list, and obviously that can end badly.  This may be fixed
+/// at a later time.
+struct fixed_buffer_producer
+{
+    char* begin;
+    char* end;
+
+    char* original_begin;
+
+    fixed_buffer_producer(char* begin, char* end) : begin(begin), end(end), original_begin(begin){
+        if (not (end > begin))
+            throw bt_serialize_length_error("Fixed buffer for serialization end pointer < begin pointer.");
+    }
+
+    // For now, used just for strings.  If we end up serializing large numbers as integers this
+    // will be used for that as well.  As this is designed for strings, will optimize for small
+    // integers and won't work for values > 99999.
+    size_t num_digits(size_t value)
+    {
```

Comment:
`{` on the same line here and elsewhere to match the existing style.  (Which I used because I've never been a fan of the wasted vertical line by putting it on the next line).

Reply:
habit; changing.